litho-processor/src/main/java/com/facebook/litho/specmodels/generator/BuilderGenerator.java

```diff
@@ -199,8 +216,26 @@ static TypeSpecDataHolder generateBuilder(SpecModel specModel) {
       initMethodSpec.addStatement("mRequired.clear()");
     }
 
+    MethodSpec.Builder initResTypePropDefaultsSpec = null;
+
+    if (isResResolvable) {
+      initResTypePropDefaultsSpec = MethodSpec.methodBuilder("initPropDefaults");
+
+      for (PropDefaultModel propDefault : propDefaults) {
+        if (!propDefault.isResResolvable()) continue;
+
+        initResTypePropDefaultsSpec.addStatement("this.$L.$L = $L",
+            getImplMemberInstanceName(specModel), propDefault.getName(),
+            generatePropsDefaultInitializers(specModel, propDefault));
+      }
+    }
+
     propsBuilderClassBuilder.addMethod(initMethodSpec.build());
 
+    if (initResTypePropDefaultsSpec != null) {
```

Comment:
why don't you just do everything at this point and then you don't have to do the null checks?